pdp/portals/gridded_observations.py

```diff
@@ -18,13 +18,14 @@ class GriddedObservationsEnsembleLister(EnsembleMemberLister):
     def list_stuff(self, ensemble):
         dataset_names = {
             "ANUSPLIN_CDA_v2012.1": "NRCANmet 2012",
-            "SYMAP_BC_v1": "PBCmet 2010",
-            "TPS_NWNA_v1": "PNWNAmet 2015"}
+            "TPS_NWNA_v1": "PNWNAmet 2015",
+            "PCIC_BLEND_v1": "PCIC Blend 2021"}
 
         for dfv in sorted(ensemble.data_file_variables,
                           key=lambda dfv: dfv.netcdf_variable_name):
-            yield dataset_names[dfv.file.run.model.short_name],\
-                dfv.netcdf_variable_name, dfv.file.unique_id.replace('+', '-')
+            if dfv.file.run.model.short_name in dataset_names.keys(): # Avoid KeyError for datasets associated with ensemble but not yet part of dataset_names
```

Comment:
> Avoid KeyError for datasets associated with ensemble but not yet part of dataset_names

How would that happen?

Reply:
The production instances query the ensemble when the page is loaded to get the datasets associated with it, so during the period where the instances don't have the new data in the `dataset_names` dictionary, then `dataset_names[dfv.file.run.model.short_name]` will throw a KeyError once the for loop looks at the new data record in the ensemble.

Reply:
So, in this case, the production instances don't have `PCIC_BLEND_v1` in the `dataset_names` dictionary right now. If I were to add the PCIC Blend datasets to the `gridded-obs-met-data` ensemble right now, the for loop would eventually look at `PCIC_BLEND_v1`, which would throw a KeyError when trying to run `yield dataset_names[dfv.file.run.model.short_name]`.